eZ/Publish/Core/Persistence/Cache/Tests/AbstractCacheHandlerTest.php

```diff
@@ -40,15 +49,43 @@ final public function testUnCachedMethods(string $method, array $arguments, arra
         $this->persistenceHandlerMock
             ->expects($this->once())
             ->method($handlerMethodName)
-            ->will($this->returnValue($innerHandler));
+            ->willReturn($innerHandler);
 
         $innerHandler
             ->expects($this->once())
             ->method($method)
             ->with(...$arguments)
-            ->will($this->returnValue($returnValue));
+            ->willReturn($returnValue);
 
         if ($tags || $key) {
+            if ($tagGeneratingArguments) {
+                $callsCount = count($tagGeneratingArguments);
+
+                $this->cacheIdentifierGeneratorMock
+                    ->expects(!empty($tagGeneratingArguments) ? $this->exactly($callsCount) : $this->never())
+                    ->method('generateTag')
+                    ->withConsecutive(...$tagGeneratingArguments)
+                    ->willReturnOnConsecutiveCalls(...$tags);
+            }
+
+            if ($keyGeneratingArguments) {
+                $callsCount = count($keyGeneratingArguments);
+
+                if (is_array($key)) {
+                    $this->cacheIdentifierGeneratorMock
+                        ->expects(!empty($keyGeneratingArguments) ? $this->exactly($callsCount) : $this->never())
```

Comment:
`count` equal to 0 is the same as declaring that the function will never be called. And the variable is always available, as it's created just before entering this block.
```suggestion
                        ->expects($this->exactly($callsCount))
```